### ISSUE TITLE
Cron module fails without error message

### ISSUE THREAD

**Issue by jackfarzan** [ansible/ansible#83223]
### Summary

When I try to provision a cron job with the cron module, the task fails but doesn't give an error message.

### Issue Type

Bug Report

### Component Name

cron

### Ansible Version

```console
$ ansible --version                                                                                                                                                           
ansible [core 2.16.5]
  config file = None
  configured module search path = ['/home/jenkins/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/dist-packages/ansible
  ansible collection location = /home/jenkins/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.3
  libyaml = True
```

### Configuration

```console
$ ansible-config dump --only-changed -t all                                                                                                                                   
WARNING: terminal is not fully functional
Press RETURN to continue 
CONFIG_FILE() = None
```

### OS / Environment

MacOS 13.2.1

### Steps to Reproduce

```yaml (paste below)
    - name: "Implement daily tmp removal cron"
      ansible.builtin.cron:
        name: "rm tmp dirs"
        minute: "5"
        hour: "1"
        job: "rm -rf /Users/my-user/tmp-*"
```

### Expected Results

I expect cron to be updated, or to return a useful error message. 

### Actual Results

```console
fatal: [x.x.x.x]: FAILED! => changed=false 
  invocation:
    module_args:
      backup: false
      cron_file: null
      day: '*'
      disabled: false
      env: false
      hour: '1'
      insertafter: null
      insertbefore: null
      job: rm -rf /Users/my-user/tmp-*
      minute: '5'
      month: '*'
      name: rm tmp dirs
      special_time: null
      state: present
      user: null
      weekday: '*'
  msg: ''
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by flowerysong** [comment]
The only code path I see that could plausibly result in this is https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cron.py#L327-L328

Does changing that line to `self.module.fail_json(msg=err, stdout=out, rc=rc)` provide any useful information?

**Comment by jackfarzan** [comment]
Unfortunately not:
```
fatal: [10.32.239.40]: FAILED! => changed=false 
  invocation:
    module_args:
      backup: false
      cron_file: null
      day: '*'
      disabled: false
      env: false
      hour: '1'
      insertafter: null
      insertbefore: null
      job: rm -rf /Users/my-user/tmp-*
      minute: '5'
      month: '*'
      name: rm tmp dirs
      special_time: null
      state: present
      user: null
      weekday: '*'
  msg: ''
  rc: -9
  stdout: ''
  stdout_lines: <omitted>
  ```
  I'm guessing it's crontab itself that isn't returning any useful error.

**Comment by jackfarzan** [comment]
Having the rc is a little bit more useful than absolutely nothing though - can I make a PR?

**Comment by mkrizek** [comment]
> Having the rc is a little bit more useful than absolutely nothing though - can I make a PR?

Sure, thanks!

**Comment by jackfarzan** [comment]
PR opened!

**Comment by Akasurde** [comment]
closed by #83290
